common/persistence/client/factory.go

```diff
@@ -119,15 +126,17 @@ var storeTypes = []storeType{
 // given configuration. In addition, all objects will emit metrics automatically
 func NewFactory(
 	cfg *config.Persistence,
+	abstractDataStoreFactory AbstractDataStoreFactory,
```

Comment:
instead of adding another arg to this constructor, can we add a member to config.Persistence called `CustomDataStoreFactory` for which no value can be specified via config i.e. yaml: "-"

Reply:
That's an option, I didn't want to add it to the config because logically it's not a config but rather an implementation, so although code will be a bit cleaner it would also be a bit more confusing doing it that way.

Reply:
that's fine by me.. in that case, consider making the AbstractDataStoreFactory an optional arg to the constructor in persistence.NewFactory.. [not a blocker ]